diff --git a/Makefile.system b/Makefile.system
index bada954..ade9b18 100644
--- a/Makefile.system
+++ b/Makefile.system
@@ -896,8 +896,8 @@ BINARY_DEFINED = 1
 endif
 
 ifeq ($(ARCH), loongarch64)
-CCOMMON_OPT += -march=loongarch64 -mabi=lp64
-FCOMMON_OPT += -march=loongarch64 -mabi=lp64
+CCOMMON_OPT += -march=loongarch64 -mabi=lp64d
+FCOMMON_OPT += -march=loongarch64 -mabi=lp64d
 endif
 
 endif
